### PR TITLE
get_task_instance_try_details API returns TaskInstanceHistory schema

### PR DESCRIPTION
**Pull request by kandharvishnu**
these API requests 
- get_task_instance_try_details
- get_mapped_task_instance_try_details
- get_task_instance_tries
- get_mapped_task_instance_tries
 
are actually returning TaskInstanceHistory

---
Airflow API Docs mentions that the [get_task_instance_try_details](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/get_task_instance_try_details) return `task_instance` schema
image

But API actually returns `TaskInstanceHistory` schema

image

[get_task_instance_try_details method](https://github.com/apache/airflow/blob/0b59ca372f6b90994047ada636c17affa38847f8/airflow/api_connexion/endpoints/task_instance_endpoint.py#L768) also returns the `TaskInstanceHistory`

image

### CODE REVIEW

File: airflow/api_connexion/openapi/v1.yaml
Comment (pierrejeambrun) [inline airflow/api_connexion/openapi/v1.yaml]: I don't think there is an `execution_date` on the `TaskInstanceHistorySchema`.

**Review by pierrejeambrun — COMMENTED** [review]
Beside the small comment, looking good.

Maybe @ephraimbuddy can double check ?

**Comment by ephraimbuddy** [comment]
There's no difference between TI and TI History except for loading related objects. It makes me wonder if we need this PR

**Comment by kandharvishnu** [comment]
> There's no difference between TI and TI History except for loading related objects. It makes me wonder if we need this PR

I'm uncertain if this PR directly impacts the [API page](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/get_task_instances_batch). However, after running a comparison between the output I received from a cURL command and the results shown on the API documentation page, I noticed some differences. Due to these discrepancies, I decided to submit a PR for further review and possible updates.

**Comment by pierrejeambrun** [comment]
> There's no difference between TI and TI History except for loading related objects. It makes me wonder if we need this PR

In that case exposing a TaskInstance might be sufficient and less confusing for users of the API. Otherwise they might be intrigued by `TaskInstanceHistory`. Maybe just a comment in the code could do the trick for us ?

**Comment by kandharvishnu** [comment]
> Also there is the deprecated `execution_date` difference, maybe the TaskInstanceHistory is missing the `logical_date`/`execution_date` ?

Additionally, I noticed that the following columns are missing in `TaskInstanceHistory` when compared with `TaskInstance`:

- `sla_miss`
- `rendered_map_index`
- `rendered_fields`
- `trigger`
- `triggerer_job`
- `note`

**Comment by ephraimbuddy** [comment]
> > Also there is the deprecated `execution_date` difference, maybe the TaskInstanceHistory is missing the `logical_date`/`execution_date` ?
> 
> Additionally, I noticed that the following columns are missing in `TaskInstanceHistory` when compared with `TaskInstance`:
> 
> * `sla_miss`
> * `rendered_map_index`
> * `rendered_fields`
> * `trigger`
> * `triggerer_job`
> * `note`

Those are the differences. We can't be able to have those in TI history.

**Comment by pierrejeambrun** [comment]
> Those are the differences. We can't be able to have those in TI history.

As there are some differences I think we need to actually fix the spec with the actual response returned. Otherwise client might expect one of those fields `trigger, etc..` while those are missing.

**Review by pierrejeambrun — APPROVED** [review]
I recommend to wait for @ephraimbuddy second opinion before merging that. But looks good to me.

**Review by ephraimbuddy — APPROVED** [review]

**Pull request merged**
